src/lib/ast.rs

```diff
@@ -35,34 +72,39 @@ pub enum Shared<T /*: std::fmt::Debug + Clone + PartialEq + Eq*/> {
 pub type FuncSort = Shared<SharedSort>;
 
 pub type TypId = Id;
+pub type TypId_ = Loc<Box<TypId>>;
 
-pub type Decs = Delim<Dec>;
-pub type Cases = Delim<Case>;
+pub type Decs = Delim<Dec_>;
+pub type Cases = Delim<Case_>;
 
 pub type Prog = Decs;
 
+pub type Dec_ = Loc<Box<Dec>>;
```

Comment:
I wonder if reading `Loc<Dec>` is simpler, where it would imply the box?

See my earlier comment about this question, near the def for `Loc`.

Reply:
Makes sense for the AST but adds a bunch of extra overhead to the lexer. If this is enough of a QoL issue, we could maybe add a standard type alias called `LB<_>` or similar. 

We could also add some specialized sugar functions by adding and `impl<X> for Loc<Box<X>>`, which makes it seem okay to have the optional heap allocation from an abstraction standpoint.